test/integration/testPutGet.js

```diff
@@ -657,3 +658,329 @@ describe('PUT GET test with GCS_USE_DOWNSCOPED_CREDENTIAL', function ()
+
+    // Create two temp file with specified file extension
+    tmpFile1 = tmp.fileSync({ postfix: 'gz' });
+    tmpFile2 = tmp.fileSync({ postfix: 'gz' });
+    // Write row data to temp file
+    fs.writeFileSync(tmpFile1.name, ROW_DATA);
+    fs.writeFileSync(tmpFile2.name, ROW_DATA);
+
+    var putQuery1 = `PUT file://${tmpFile1.name} @${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+    // Windows user contains a '~' in the path which causes an error
+    if (process.platform == "win32")
+    {
+      var fileName = tmpFile1.name.substring(tmpFile1.name.lastIndexOf('\\'));
+      putQuery1 = `PUT file://${process.env.USERPROFILE}\\AppData\\Local\\Temp\\${fileName} @${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+    }
+
+    var putQuery2 = `PUT file://${tmpFile2.name} @${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+    // Windows user contains a '~' in the path which causes an error
+    if (process.platform == "win32")
+    {
+      var fileName = tmpFile2.name.substring(tmpFile2.name.lastIndexOf('\\'));
+      putQuery2 = `PUT file://${process.env.USERPROFILE}\\AppData\\Local\\Temp\\${fileName} @${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+    }
+
+    // Create a tmp folder for downloaded files
+    var tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'get'));
+    var fileSize;
+
+    var getQuery = `GET @${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME} file://${tmpDir}`;
+    // Windows user contains a '~' in the path which causes an error
+    if (process.platform == "win32")
+    {
+      var dirName = tmpDir.substring(tmpDir.lastIndexOf('\\') + 1);
+      getQuery = `GET @${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME} file://${process.env.USERPROFILE}\\AppData\\Local\\Temp\\${dirName}`;
+    }
+
+    async.series(
+      [
+        function (callback)
+        {
+          // Create temp table
+          testUtil.executeCmd(connection, createTable, callback);
+        },
+        function (callback)
+        {
+          // Upload file
+          var statement = connection.execute({
+            sqlText: putQuery1,
+            complete: function (err, stmt, rows)
+            {
+              var stream = statement.streamRows();
+              stream.on('error', function (err)
+              {
+                done(err);
+              });
+              stream.on('data', function (row)
+              {
+                fileSize = row.targetSize;
+                // Check the file is correctly uploaded
+                assert.strictEqual(row['status'], UPLOADED);
+                // Check the target encoding is correct
+                assert.strictEqual(row['targetCompression'], 'GZIP');
+              });
+              stream.on('end', function (row)
+              {
+                callback();
+              });
+            }
+          });
+        },
+        function (callback)
+        {
+          // Upload file
+          var statement = connection.execute({
```

Comment:
this execution is the same as the one above but put's different file? can we create one method for put?